src/node_worker.cc

```diff
@@ -816,6 +816,100 @@ void Worker::Unref(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+void Worker::GetHeapStatistics(const FunctionCallbackInfo<Value>& args) {
+  Worker* w;
+  ASSIGN_OR_RETURN_UNWRAP(&w, args.This());
+
+  v8::HeapStatistics heap_stats;
+  w->isolate_->GetHeapStatistics(&heap_stats);
+
+  auto* isolate = args.GetIsolate();
+  Local<Context> currentContext = isolate->GetCurrentContext();
+
+  Local<Object> stats = Object::New(isolate);
+
+  stats
```

Comment:
Instead of listing these out manually, the `HEAP_STATISTICS_PROPERTIES` X macro could be used
https://github.com/nodejs/node/blob/91d8a524ada001103a2d1c6825ca17b8393c183f/src/node_v8.cc#L57-L71